store/tikv/coprocessor.go

```diff
@@ -628,6 +631,58 @@ func (worker *copIteratorWorker) handleTaskOnce(bo *Backoffer, task *copTask, ch
 	return worker.handleCopResponse(bo, resp.Cop, task, ch, nil)
 }
 
+const (
+	minLogBackoffTime       = 100
+	minLogKVProcessTime     = 100
+	minLogKVWaitTime        = 200
+	minLogScanProcessedInfo = 100000
+	minLogScanTotalInfo     = 200000
+)
+
+func (worker *copIteratorWorker) logTimeCopTask(costTime time.Duration, task *copTask, bo *Backoffer, resp *tikvrpc.Response) {
+	logStr := fmt.Sprintf("[TIME_COP_TASK] resp_time:%s txn_start_ts:%d region_id:%d store_addr:%s", costTime, worker.req.StartTs, task.region.id, task.storeAddr)
+	if bo.totalSleep > minLogBackoffTime {
+		backoffTypes := strings.Replace(fmt.Sprintf("%v", bo.types), " ", ",", -1)
+		logStr += fmt.Sprintf(" backoff_ms:%d backoff_types:%s", bo.totalSleep, backoffTypes)
+	}
+	var detail *kvrpcpb.ExecDetails
+	if task.cmdType == tikvrpc.CmdCopStream {
+		detail = resp.CopStream.ExecDetails
+	} else {
+		detail = resp.Cop.ExecDetails
+	}
+	if detail != nil {
```

Comment:
There are so many `if` in `if`..  Could we `log.Info` through `defer` function, and return directly `if detail==nil`?

Reply:
I think defer is used for release resource.
I looks weird if the normal execution code is done in defer.

Reply:
Agree with Yuxing.